[tools] Build img2simg_host and simg2img_host for hybris-hal

kjokinie opened the pull request:
When doing image creation in OBS or mic, the default image creation
tools from droid-hal-tools package don't always work if the device
requires special version of those tools.

Let's build all the tools: img2simg, simg2img, and mkbootimg always
with hybris-hal, so they can easily be packaked in droid-hal-device
to a separate device specific tools package. Please note mkbootimg
is already required by hybris-boot, so no special dependency was
added for it.

Signed-off-by: Kalle Jokiniemi <kalle.jokiniemi@jolla.com>

sledges commented on Android.mk:
```diff
@@ -234,5 +236,5 @@ HYBRIS_UPDATER_UNPACK := $(LOCAL_BUILD_MODULE)
 
 .PHONY: hybris-hal
-hybris-hal: bootimage hybris-updater-unpack hybris-updater-script hybris-recovery hybris-boot linker init libc adb adbd libEGL libGLESv2 servicemanager logcat updater
+hybris-hal: bootimage hybris-updater-unpack hybris-updater-script hybris-recovery hybris-boot $(IMG2SIMG) $(SIMG2IMG) linker init libc adb adbd libEGL libGLESv2 servicemanager logcat updater
```
It should be img2simg_host simg2img_host instead

sledges commented on Android.mk:
```diff
@@ -72,7 +72,9 @@ $(error There should be a one and only one device entry for HYBRIS_BOOT_PART and
-MKBOOTIMG := mkbootimg
+MKBOOTIMG := $(HOST_OUT_EXECUTABLES)/mkbootimg$(HOST_EXECUTABLE_SUFFIX)
+SIMG2IMG := $(HOST_OUT_EXECUTABLES)/simg2img$(HOST_EXECUTABLE_SUFFIX)
+IMG2SIMG := $(HOST_OUT_EXECUTABLES)/img2simg$(HOST_EXECUTABLE_SUFFIX)
```
two lines above are not used here, thus not needed

kjokinie commented:
What's new:

* Fixed commit description to tell what we really are after

* Removed changes to MKBOOTIMG etc macros as those changes were not needed

sledges commented:
LGTM, merge with https://github.com/mer-hybris/droid-hal-device/pull/88
